src/main/java/trackitnus/model/ModelManager.java

```diff
@@ -323,6 +323,16 @@ public Index getLessonIndex(Lesson lesson) throws CommandException {
         return Index.fromZeroBased(index);
     }
 
+    @Override
+    public Index getModuleIndex(Module module) throws CommandException {
+        ObservableList<Module> moduleList = getFilteredModuleList();
+        int index = moduleList.indexOf(module);
+        if (index == -1) {
+            throw new CommandException(Messages.MESSAGE_MODULE_DOES_NOT_EXIST);
+        }
```

Comment:
Sorry I just realised, should you update the predicate to show all Modules? Or doesn't matter. In task and lesson we will predicate first